randovania/gui/widgets/select_preset_widget.py

```diff
@@ -316,17 +325,40 @@ def on_preset_changed(self, preset: VersionedPreset | None):
                     description += preset_describer.merge_categories(preset_describer.describe(raw_preset))
 
             except InvalidPreset as e:
-                if not isinstance(e.original_exception, UnsupportedVersion):
-                    logging.exception(f"Invalid preset for {preset.name}")
+                if isinstance(e.original_exception, UnsupportedVersion):
+                    exception_desc = f"<p>{e.original_exception}</p>"
+                else:
+                    logging.warning(f"Invalid preset for {preset.name}")
+                    exception_desc = "<pre>{}</pre>".format(
+                        "\n".join(traceback.format_exception(e.original_exception))
+                    )
+
                 description = (
-                    f"Preset {preset.name} can't be used as it contains the following error:"
-                    f"\n{e.original_exception}\n"
-                    f"\nPlease open edit the preset file with id {preset.uuid} manually or delete this preset."
+                    f"<p>Preset {preset.name} can't be used as it contains errors.</p>"
+                    f"<p>Please edit the file named <a href='open-preset://{preset.uuid}'>{preset.uuid}</a> manually "
+                    f"or delete this preset.</p>"
+                    f"{exception_desc}"
                 )
 
         self.create_preset_description.setText(description)
+        self._can_generate = can_generate
         self.CanGenerate.emit(can_generate)
 
+    def _on_click_create_preset_description(self, link: str):
+        info = re.match(r"^open-preset://([0-9a-f\-]{36})$", link)
```

Comment:
might as well `re.compile` this for reuse imo

Reply:
I doubt this search is done more than once per app, and most of the time none. Better performance doing it here.